Helper/AdyenOrderPayment.php

```diff
@@ -84,92 +89,108 @@ public function __construct(
+     * @param string $currency
+     * @return Payment
+     * @throws AlreadyExistsException
+     */
+    public function refreshPaymentCaptureStatus(Payment $adyenOrderPayment, string $currency): Payment
+    {
+        $totalCapturedCents = $this->adyenDataHelper->formatAmount($adyenOrderPayment->getTotalCaptured(), $currency);
+        $amountCents = $this->adyenDataHelper->formatAmount($adyenOrderPayment->getAmount(), $currency);
 
-        if ($totalAmountAuthorized === $orderTotalAmount) {
-            return true;
+        if ($totalCapturedCents < $amountCents) {
+            $captureStatus = OrderPaymentInterface::CAPTURE_STATUS_PARTIAL_CAPTURE;
+        } else {
+            $captureStatus = OrderPaymentInterface::CAPTURE_STATUS_MANUAL_CAPTURE;
         }
 
-        return false;
+        $adyenOrderPayment->setCaptureStatus($captureStatus);
+        $this->orderPaymentResourceModel->save($adyenOrderPayment);
+
+        return $adyenOrderPayment;
     }
 
     /**
-     * Check if ANY adyen_order_payment linked to this order requires a manual capture
+     * Check if the full amount of the order has been finalized, either automatically by checking the
+     * adyen_order_payment entries, or manually by checking the invoices
      *
      * @param Order $order
-     * @param string $status
      * @return bool
      */
-    public function hasOrderPaymentWithCaptureStatus(Order $order, string $status): bool
+    public function isFullAmountFinalized(Order $order): bool
     {
-        $requireManualCapture = false;
-        $payment = $order->getPayment();
-        $adyenOrderPayments = $this->orderPaymentResourceModel->getLinkedAdyenOrderPayments($payment->getEntityId());
+        $invoiceAmountCents = 0;
+        $autoAdyenOrderPayments = $this->orderPaymentResourceModel->getLinkedAdyenOrderPayments(
+            $order->getPayment()->getEntityId(),
+            [OrderPaymentInterface::CAPTURE_STATUS_AUTO_CAPTURE]
+        );
 
-        foreach ($adyenOrderPayments as $adyenOrderPayment) {
-            if ($adyenOrderPayment[OrderPaymentInterface::CAPTURE_STATUS] === $status) {
-                $requireManualCapture = true;
+        // If the full amount has been automatically captured
+        if ($this->compareAdyenOrderPaymentsAmount($order, $autoAdyenOrderPayments)) {
+            return true;
+        }
+
+        $invoices = $order->getInvoiceCollection();
+        /** @var Order\Invoice $invoice */
+        foreach ($invoices as $invoice) {
+            if ($this->invoiceHelper->isFullInvoiceAmountManuallyCaptured($invoice)) {
+                $invoiceAmountCents += $this->adyenDataHelper->formatAmount(
+                    $invoice->getGrandTotal(),
+                    $invoice->getOrderCurrencyCode()
+                );
             }
         }
 
-        return $requireManualCapture;
+        $orderAmountCents = $this->adyenDataHelper->formatAmount($order->getGrandTotal(), $order->getOrderCurrencyCode());
```

Comment:
Should we use the charged currency helper to get the order data?

Reply:
See https://github.com/Adyen/adyen-magento2/pull/1297#discussion_r777335221